oss/build_whl.sh

```diff
@@ -81,6 +77,15 @@ function main() {
   popd
 
   echo $(date) : "=== Output wheel file is in: ${DEST}"
+
+  # Install ArrayRecord from the wheel and run smoke tests.
+  # TF is not available on Python 3.13 and above.
+  if (( "${PYTHON_MINOR_VERSION}" < 13 )); then
+    $PYTHON_BIN -m pip install --find-links=/tmp/grain/all_dist --pre array-record
```

Comment:
/tmp/grain/all_dist -> /tmp/array_record/all_dist

I think in the workflow run you shared it still installed AR from pypi: https://github.com/mtsokol/array_record/actions/runs/16147213833/job/45569132362#step:7:7181

Reply:
I think this is something similar to: https://github.com/apache/arrow/issues/40088

Reply:
yeah, I also think this is because of the protobuf version. The problem here is not so much in grain, but tensorflow -- updating protobuf in grain is easy, but if both AR and Grain have higher protobuf version than TF, then import with TF will fail the same way. This is how we found out in the first place about this -- TF is not a dependency, but is often installed in the same ecosystem.

I think protobuf version was updated in TF main: https://github.com/tensorflow/tensorflow/pull/92241. So now, hopefully with the next release of TF we can update it too.

Reply:
Could we try solve this using TensorFlow nightly in our test? Some in https://pypi.org/project/tf-nightly/#history should already use newer protobuf, right?

Reply:
yes for tests, but our users wouldn't be using nightly most likely: https://github.com/google-research/kauldron/blob/8537718c703ab606a545eee6daf89238d8def55f/pyproject.toml#L41